marketplace/src/main/assemble/assembly.xml

```diff
@@ -53,26 +54,68 @@
           <artifact scope="provided" />
           <artifact type="pom" />
         </excludes>
-      </artifact:set>
+      </artifact:dependencies>
     </copy>
     <!-- Third-party libraries -->
     <copy todir="${outdir}/nxr/lib" overwrite="true">
-      <artifact:set>
+      <artifact:dependencies key="org.nuxeo.ecm.platform:nuxeo-agenda-core:::">
         <includes>
           <artifact groupId="!org.nuxeo*" />
           <artifact groupId="org.nuxeo.lib*" />
         </includes>
         <excludes>
           <artifact scope="test" />
         </excludes>
-      </artifact:set>
+      </artifact:dependencies>
     </copy>
     <nx:rmdups dir="${outdir}/nxr/lib" />
+    <!-- TODO What should be unzipped / zipped before `nx:zipdiff`? Not sure what I did is correct. -->
```

Comment:
(always associate a TODO with a JIRA ticket: `<!-- TODO NXP-21383 ... -->`)

What you did looks fine.
`nx:zipdiff` must be called when the target NXR is ready. The task will compare the zipped NXR with the server one in order to embed only the missing files in the package.